src/index.ts

```diff
@@ -1,18 +1,24 @@
 #!/usr/bin/env node
 
-const { execSync } = require("child_process");
-const path = require("path");
-const Discord = require("discord.js");
-const fs = require("fs-extra");
-const prompts = require("prompts");
-const validatePackageName = require("validate-npm-package-name");
+import { execSync } from "child_process";
+import { Client } from "discord.js";
+import fs from "fs-extra";
+import path from "path";
+import prompts from "prompts";
 
-const appDirectory = path.join(__dirname, "app");
-const appPackage = require(path.join(appDirectory, "package.json"));
-const appToken = { token: "DISCORD_BOT_TOKEN_PLACEHOLDER" };
+import interfaces from "./declarations/interfaces";
+import types from "./declarations/types";
 
-const utilityPackage = require(path.join(__dirname, "package.json"));
-const utilityNameAndVersion = `${utilityPackage.name} v${utilityPackage.version}`;
+const validatePackageName: interfaces.ValidationFunction = require("validate-npm-package-name");
+
+const appDirectory: string = "app";
+const appPackage: types.Package = require(path.resolve(
+  path.join(appDirectory, "package.json")
+));
+const appToken: { token: string } = { token: "DISCORD_BOT_TOKEN_PLACEHOLDER" };
```

Comment:
maybe this token type can also be moved to our `types` file? not sure what "threshold" we should adopt for when to declare types inline vs have it separated into `types`, currently thinking having primitive types be inline while anything more complex can be in `types. lmk your thoughts.

Reply:
I was thinking of it but then again, it is basically only used once through out the entirety of the script I believe.

Reply:
hmm i'm not sure i can rationalize `used only once` as our criteria. e.g. say a more complex type is used only once, where do we draw the line? arbitrarily say 1 or 2 or _n_ properties qualifies it for inline?

Reply:
Well, it does only have one property. And, anyways, it isn't even useful to keep a token inside an object, we could just make it a string.

Reply:
that works.
```
const token: string = "DISCORD_BOT_TOKEN_PLACEHOLDER";

and then way below where we write the token just `{ token }` in the `JSON.stringify`

Reply:
have to also update the `initial` token value in the `questions` array